charts/tekton-apps/templates/wordpress/argocd_applications.yaml

```diff
@@ -22,161 +17,171 @@ metadata:
-          {{ $k }}: {{ $v | quote}}
-          {{- end }}
+        {{- if .nodeSelector }}
+          {{- toYaml .nodeSelector | nindent 10 }}
         {{- else }}
-        nodeSelector:
           tech_stack: php
           pvc: "true"
         {{- end }}
 
-        {{ if hasKey $component.wordpress "podSecurityContext" }}
-        podSecurityContext:
-          {{- range $k, $v := $component.wordpress.podSecurityContext }}
-          {{ $k }}: {{ $v }}
-          {{- end }}
-        {{- else }}
+        {{- if .podSecurityContext }}
         podSecurityContext:
-          enabled: true
-          fsGroup: 1000
+          {{- toYaml .podSecurityContext | nindent 10 }}
         {{- end }}
 
-        {{ if hasKey $component.wordpress "containerSecurityContext" }}
+        {{- if .containerSecurityContext }}
         containerSecurityContext:
-          {{- range $k, $v := $component.wordpress.containerSecurityContext }}
-          {{ $k }}: {{ $v }}
-          {{- end }}
-        {{- else }}
-        containerSecurityContext:
-          enabled: true
-          runAsUser: 1000
+          {{- toYaml .containerSecurityContext | nindent 10 }}
         {{- end }}
 
-        {{ if ternary $component.wordpress.ci "true" (hasKey $component.wordpress "ci") }}
+        {{ if or .ci (not (hasKey . "ci")) -}}
         initContainers:
           - name: install
-            image: jnewland/git-and-stuff
+            image: bitnami/git
             imagePullPolicy: Always
+            securityContext:
+              runAsUser: {{ (.containerSecurityContext).runAsUser | default 1001 }}
+            env:
+              - name: HOME
+                value: /home/git
             command:
               - bash
               - -c
               - |
-                git -c core.sshCommand="ssh -i ~/.ssh/id_rsa" clone {{ $component.wordpress.repository_ssh_url }} /tmp/wordpress
-                cp -Rf /tmp/wordpress/wp-content/* /bitnami/wordpress/wp-content/
-                cd /tmp/wordpress
-                git_branch=`git rev-parse --abbrev-ref HEAD`
-                git_log=`git log --format="%h - by %an - %ae - %s" HEAD~1..HEAD`
-                msg="[$git_branch] $git_log"
-                printf "%0.s-" $(seq 1 ${#msg})
-                echo
-                echo "$msg"
-                printf "%0.s-" $(seq 1 ${#msg})
-                echo
-                ls -la /tmp/wordpress/wp-content
-                echo "Folder wp-content copied from git repo into bitnami/wodpress"
-                echo "Done INIT"
+                set -o errexit
+                set -o nounset
+                set -o pipefail
+                #set -o xtrace
+                WP_ROOT=/bitnami/wordpress
+                COPY_WP_CONF="{{ and .repositoryUseWPConfig (not .existingWordPressConfigurationSecret) | default "false"}}"
+                URL="{{ .repositorySshUrl | required "Missed .wordpress.repositorySshUrl option"}}"
```

Comment:
I've tested variables with minimal and maximum filled values. Empty vars is part of condition
```
helm template . -s templates/wordpress/argocd_applications.yaml -f values-all.yaml
helm template . -s templates/wordpress/argocd_applications.yaml -f values-minimal.yaml
```

Reply:
@MikhailKorol-saritasa 
Misha, my request was different. Instead having `| required` calls in different places of the file - make validation at the top of the file, so all your required variables are validated together in one place.
See for example
charts/opsgenie-heartbeat/templates/NOTES.txt

That way it's much easier to see what is needed and keep things under control.
You don't need to export all your validation into NOTES.txt file as in  `charts/opsgenie-heartbeat/templates/NOTES.txt` since you validate only specific case related to wordpress. As a result - just move all required checks at the top of this file.

Reply:
@dmitry-mightydevops 
I strongly disagree with that. 
Group validation is good when we are checking reusable values that spreaded along a file. In our case we have single usage for variable and `| required` does job done. Best example is using `default`. We have reusable vars on top and single usage vars across a file. All of that makes template relieble and easy to read and support.